inline.go

```diff
@@ -794,10 +793,6 @@ func autoLink(p *parser, out *bytes.Buffer, data []byte, offset int) int {
 
 			bufEnd--
 		}
-
-		if openDelim == 0 {
-			linkEnd--
-		}
```

Comment:
Won't this break this case? `foo (http://en.wikipedia.org/wiki/Disambiguation_(disambiguation)) bar`

Autolink detector should recognize that the outer parentheses do not belong to the link. Can you add this case to the tests?